app/state_machines/efile_submission_state_machine.rb

```diff
@@ -129,4 +145,17 @@ def self.send_mixpanel_event(efile_submission, event_name, data: {})
       data: data,
     )
   end
+
+  def self.source_to_state(source)
+    # Note that `state_info` here will contain a two member array of key and
+    # value from the original hash. will look something like
+    # ['az', {intake_class: "StateFileAzIntake"]
+    US_STATE_MAPPING_CACHE[source] ||= StateFile::StateInformationService::STATES_INFO.find do |state_info|
+      state_info[1][:intake_class].name == source
+    end[0]
+  end
+
+  def self.currently_stopped_states
+    ENV.fetch('HOLD_OFF_EFILE_SUBMISSIONS_FOR_STATES', '').split.map(&:downcase)
```

Comment:
[dust] if we're chasing runtime efficiency, downcasing the source string before splitting saves us a loop.

```suggestion
    ENV.fetch('HOLD_OFF_EFILE_SUBMISSIONS_FOR_STATES', '').downcase.split
```